diff --git a/src/components/article/Comment.tsx b/src/components/article/Comment.tsx
index 4846c50..9fce363 100644
--- a/src/components/article/Comment.tsx
+++ b/src/components/article/Comment.tsx
@@ -20,12 +20,13 @@ const Comment = ({ comments, slug }: ICommentProps) => {
   const onPostComment = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     const { body } = newComment;
+    if (!body.trim()) return;
     createCommentMutation.mutate(
       { body, slug },
       {
-        onSuccess: (_) => {
-          setNewComment({ body: '', slug });
-          queryClient.invalidateQueries({ queryKey: [QUERY_COMMENTS_KEY] });
+        onSuccess: () => {
+          setNewComment({ body: '' });
+          queryClient.invalidateQueries({ queryKey: [QUERY_COMMENTS_KEY, slug] });
         },
       },
     );
@@ -57,7 +58,7 @@ const Comment = ({ comments, slug }: ICommentProps) => {
         </div>
         <div className="card-footer">
           <img src={data.image} className="comment-author-img" alt="comment-author" />
-          <button type="submit" className="btn btn-sm btn-primary">
+          <button type="submit" className="btn btn-sm btn-primary" disabled={createCommentMutation.isLoading}>
             Post Comment
           </button>
         </div>
