diff --git a/src/components/feed/FeedList.tsx b/src/components/feed/FeedList.tsx
index e357a2b..4288cd8 100644
--- a/src/components/feed/FeedList.tsx
+++ b/src/components/feed/FeedList.tsx
@@ -30,7 +30,7 @@ const FeedList = ({ articlesInfo, page, setPage }: IFeedListProps) => {
       <nav>
         <ul className="pagination">
           {articlesCount > UNIT_PER_PAGE &&
-            generateOneToNArray(Math.floor(articlesCount / UNIT_PER_PAGE)).map((value) => (
+            generateOneToNArray(Math.ceil(articlesCount / UNIT_PER_PAGE)).map((value) => (
               <li
                 key={value}
                 role="presentation"
diff --git a/src/pages/HomePage.tsx b/src/pages/HomePage.tsx
index 81376c8..1709505 100644
--- a/src/pages/HomePage.tsx
+++ b/src/pages/HomePage.tsx
@@ -11,6 +11,24 @@ const HomePage = () => {
   const [selectedTag, setSelectedTag] = useState('');
   const [articlesInfo, tagsInfo] = useGetArticlesQueries(isGlobal, page, selectedTag);
 
+  const onSelectGlobalFeed = () => {
+    setIsGlobal(true);
+    setSelectedTag('');
+    setPage(1);
+  };
+
+  const onSelectYourFeed = () => {
+    setIsGlobal(false);
+    setSelectedTag('');
+    setPage(1);
+  };
+
+  const onSelectTag = (tag: string) => {
+    setIsGlobal(true);
+    setSelectedTag(tag);
+    setPage(1);
+  };
+
   return (
     <div className="home-page">
       <div className="banner">
@@ -27,19 +45,13 @@ const HomePage = () => {
               <ul className="nav nav-pills outline-active">
                 {isLogin && (
                   <li className="nav-item">
-                    <Link
-                      className={`nav-link ${isGlobal ? '' : 'active'}`}
-                      to="/"
-                      onClick={() => {
-                        setIsGlobal(false);
-                      }}
-                    >
+                    <Link className={`nav-link ${isGlobal ? '' : 'active'}`} to="/" onClick={onSelectYourFeed}>
                       Your Feed
                     </Link>
                   </li>
                 )}
                 <li className="nav-item">
-                  <Link className={`nav-link ${isGlobal ? 'active' : ''}`} to="/" onClick={() => setIsGlobal(true)}>
+                  <Link className={`nav-link ${isGlobal ? 'active' : ''}`} to="/" onClick={onSelectGlobalFeed}>
                     Global Feed
                   </Link>
                 </li>
@@ -54,14 +66,7 @@ const HomePage = () => {
 
               <div className="tag-list">
                 {tagsInfo.data.map((tag: string) => (
-                  <Link
-                    to="/"
-                    key={tag}
-                    className="tag-pill tag-default"
-                    onClick={() => {
-                      setSelectedTag(tag);
-                    }}
-                  >
+                  <Link to="/" key={tag} className="tag-pill tag-default" onClick={() => onSelectTag(tag)}>
                     {tag}
                   </Link>
                 ))}
