diff --git a/src/components/ErrorFallback.tsx b/src/components/ErrorFallback.tsx
index 7786b1d..1413e93 100644
--- a/src/components/ErrorFallback.tsx
+++ b/src/components/ErrorFallback.tsx
@@ -17,10 +17,7 @@ const ErrorFallback = ({ resetErrorBoundary }: IErrorFallbackProps) => {
         <button
           type="button"
           className="btn btn-outline-primary"
-          onClick={() => {
-            navigate('/', { replace: true });
-            window.location.reload();
-          }}
+          onClick={() => navigate('/', { replace: true })}
         >
           Go Home
         </button>
diff --git a/src/repositories/apiClient.ts b/src/repositories/apiClient.ts
index f42976f..4ce5a7d 100644
--- a/src/repositories/apiClient.ts
+++ b/src/repositories/apiClient.ts
@@ -38,12 +38,13 @@ apiClient.interceptors.response.use(
   },
   (error) => {
     const { message } = error;
-    const { status, data } = error.response;
-    const { method, url } = error.config;
+    const status = error.response?.status;
+    const data = error.response?.data;
+    const method = error.config?.method;
+    const url = error.config?.url;
 
     if (status === 429) {
-      token.removeToken('ACCESS_TOKEN_KEY');
-      window.location.reload();
+      token.removeToken(ACCESS_TOKEN_KEY);
     }
 
     logOnDev(`🚨 [${method?.toUpperCase()}] ${url} | Error ${status} ${data?.message || ''} | ${message}`, error);
